Makefile

```diff
@@ -78,18 +78,20 @@ help: ## Print help for targets with comments.
 # to check for changes.
 .PHONY: $(NODE_EXE) $(NODE_G_EXE)
 
+define build_node_exe
+$(MAKE) -C out BUILDTYPE=$1 V=$(V)
 # The -r/-L check stops it recreating the link if it is already in place,
 # otherwise $(NODE_EXE) being a .PHONY target means it is always re-run.
 # Without the check there is a race condition between the link being deleted
 # and recreated which can break the addons build when running test-ci
 # See comments on the build-addons target for some more info
+if [ ! -r $@ -o ! -L $@ ]; then ln -fs out/$1/$(NODE_EXE) $@; fi
+endef
 $(NODE_EXE): config.gypi out/Makefile
```

Comment:
AFAICT `$BUILDTYPE` is already defined.
https://github.com/nodejs/node/blob/db723a102004829c5d518b57789b424faa6d07cb/Makefile#L64-L68
so this could be:
```gmake
$(NODE_EXE) $(NODE_G_EXE): config.gypi out/Makefile
  $(MAKE) -C out
  if [ ! -r $@ -o ! -L $@ ]; then ln -fs out/$(BUILDTYPE)/$(NODE_EXE) $@; fi
```
removing
`BUILDTYPE` is defined so no need to pass is explicitly
`V=$(V)` this is just 🤷‍♂️ 
and putting `$(BUILDTYPE)` in the link target